lib/bind.js

```diff
@@ -19,7 +19,16 @@ function bind(fn, context, args) {
     assert.optionalArray(args, 'args');
 
     return function boundFn() {
-        fn.apply(context || null, args || []);
+
+        // working with arguments is a bit hazardous in terms of deopt.
+        // https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#3-managing-arguments
```

Comment:
I was curious myself, as I knew `arguments` was less than performant, but didn't know exactly what usage would cause degradation. This all might be moot if/when using ES2015's array spread operator (`[...arguments]`).